modules/samples/oauth2/playground2/src/main/webapp/oauth2.jsp

```diff
@@ -483,11 +525,71 @@
+                    if (StringUtils.countMatches(idToken, ".") == 4) {
+                        // it's a JWE
+            %>
+
+            <div>
+                <form action="oauth2-access-resource.jsp" id="loginForm" method="post">
+
+                    <table class="user_pass_table" id="userPassTableJWE">
+                        <tbody>
+                        <tr>
+                            <td><label>Access Token :</label></td>
+                            <td><input id="accessToken" name="accessToken" style="width:350px"
+                                       value="<%=accessToken%>"/>
+                        </tr>
+                        <tr>
+                            <td><label>UserInfo Endpoint :</label></td>
+                            <td><input id="resource_url" name="resource_url" type="text" value="<%=userInfo%>"
+                                       style="width:350px"/>
+                        </tr>
+                        <tr>
+                            <td><label>Encrypted ID Token :</label></td>
+                            <td><textarea id="encryptedIdToken" name="idToken" style="width:450px" disabled="true">
+                                <%=idToken.trim()%>
+                            </textarea>
+                        </tr>
+                        <tr>
+                            <td><label>Client Private Key :</label></td>
+                            <td><textarea id="clientPrivateKey" name="privateKey" style="width:450px"></textarea></td>
+                            <td>
+                                <input type="submit" class="button" value="Decrypt"
+                                       onclick="decryptIdToken();return false;">
+                            </td>
+                        </tr>
+
+                        <tr>
+                            <td>
+                                <input type="submit" class="button" value="Get UserInfo">
+                            </td>
+                            <%
+                                if (isOIDCLogoutEnabled) {
+                            %>
+                            <td>
+                                <button type="button" class="button"
+                                        onclick="document.location.href='<%=(String)session.getAttribute(OAuth2Constants.OIDC_LOGOUT_ENDPOINT)%>';">
+                                    Logout
+                                </button>
+                            </td>
+                            <%
+                                }
+                            %>
+                        </tr>
+                        </tbody>
+                    </table>
+
+                </form>
+            </div>
+
+            <%
+                    } else {
+                        try {
+                            name = SignedJWT.parse(idToken).getJWTClaimsSet().getSubject();
+                        } catch (Exception e) {
+                            //ignore
```

Comment:
Shouldn't we at least log the exception?

Reply:
This was already there before me. What should be the best way? Can we log them since this is a jsp?